libraries/AP_HAL_ChibiOS/hwdef/f303-ADSB/hwdef.dat

```diff
@@ -1,48 +1,138 @@
+FLASH_SIZE_KB 256
+
+
+# order of UARTs
+#SERIAL_ORDER USART2 USART3 EMPTY USART1
+SERIAL_ORDER USART2 EMPTY EMPTY EMPTY
+
+# a LED to flash
+PA4 LED OUTPUT LOW
+
+define HAL_CAN_POOL_SIZE 6000
+
+# USART1, connected to RADAR
+#PA9  USART1_TX USART1 SPEED_HIGH
+#PA10 USART1_RX USART1 SPEED_HIGH
+
+# USART2 for ADSB
+PA2 USART2_TX USART2 SPEED_HIGH
+PA3 USART2_RX USART2 SPEED_HIGH
+
+# USART3 for Debug
+#PB10 USART3_TX USART3 SPEED_HIGH NODMA
+#PB11 USART3_RX USART3 SPEED_HIGH NODMA
+
+# spi bus for compass
+#PA5 SPI1_SCK SPI1
+#PA6 SPI1_MISO SPI1
+#PA7 SPI1_MOSI SPI1
+
+# analog input
+# PA5 VIN5 ADC1
+define HAL_USE_ADC FALSE
+define STM32_ADC_USE_ADC1 FALSE
+define HAL_DISABLE_ADC_DRIVER TRUE
+
+define HAL_NO_GPIO_IRQ
+define SERIAL_BUFFERS_SIZE 512
+define PORT_INT_REQUIRED_STACK 64
 
+# avoid timer and RCIN threads to save memory
+define HAL_NO_RCIN_THREAD
 
-######################################
-# disable features set in f303-periph
-######################################
+define HAL_USE_RTC FALSE
+define DISABLE_SERIAL_ESC_COMM TRUE
+define NO_DATAFLASH TRUE
 
-# I2C is not used
-undef PB6
-undef PB7
-undef HAL_USE_I2C
-undef STM32_I2C_USE_I2C1
-undef I2C_ORDER
-undef HAL_I2C_CLEAR_ON_TIMEOUT
+define DMA_RESERVE_SIZE 0
+define STM32_DMA_REQUIRED TRUE
 
-# no compass
-undef PB0
-undef HAL_COMPASS_MAX_SENSORS
+define PERIPH_FW TRUE
 
-# SPI1 is not used
-undef PA5
-undef PA6
-undef PA7
-undef SPI1
+# MAIN_STACK is stack of initial thread
+MAIN_STACK 0x80
 
-# disable debugger support (TODO - disable this when ready for production)
-undef PA13
-undef PA14
+# PROCESS_STACK controls stack for main thread
+PROCESS_STACK 0x800
+define HAL_DISABLE_LOOP_DELAY
 
-# no GPS
-undef GPS_MAX_RECEIVERS
-undef GPS_MAX_INSTANCES
+# enable CAN support
+PA11 CAN_RX CAN
+PA12 CAN_TX CAN
 
+# debugger support
+#uncomment these to allow support via SWD (STLINK-V3 or equivalent)
+#PA13 JTMS-SWDIO SWD
+#PA14 JTCK-SWCLK SWD
+
+define HAL_UART_MIN_TX_SIZE 256
+define HAL_UART_MIN_RX_SIZE 128
+
+define HAL_UART_STACK_SIZE 256
+
+define STORAGE_THD_WA_SIZE 256
+define IO_THD_WA_SIZE      256
+
+define HAL_NO_GCS
+define HAL_NO_LOGGING
+define HAL_NO_MONITOR_THREAD
+
+define HAL_MINIMIZE_FEATURES 0
+
+define HAL_BUILD_AP_PERIPH
+
+define HAL_DEVICE_THREAD_STACK 256
+
+define AP_PARAM_MAX_EMBEDDED_PARAM 0
+
+# use the app descriptor needed by MissionPlanner for CAN upload
+env APP_DESCRIPTOR MissionPlanner
+
+# reserve 256 bytes for comms between app and bootloader
+RAM_RESERVE_START 256
+
+# keep ROMFS uncompressed as we don't have enough RAM
+# to uncompress the bootloader at runtime
+env ROMFS_UNCOMPRESSED True
+
+# reduce the number of CAN RX Buffer
+define HAL_CAN_RX_QUEUE_SIZE 32
+
+define CAN_APP_NODE_NAME "com.matternet.trafficmonitor"
+
+# use DNA
+define HAL_CAN_DEFAULT_NODE_ID 0
+
+# ADSB enable
+define ADSB_PORT hal.uartA
+define HAL_PERIPH_ENABLE_ADSB
+#undef HAL_PERIPH_ADSB_BAUD_DEFAULT
+define HAL_PERIPH_ADSB_BAUD_DEFAULT 57600
```

Comment:
Why did the default baud decrease from 115200 to 57600?

Reply:
the pingRX ADS-B is at 57600.  From what I can see in the [datasheet](https://uavionix.com/downloads/pingrx/pingRX-Datasheet.pdf), that baud rate cannot be changed.

Reply:
@tridge do you know why HAL_PERIPH_ADSB_BAUD_DEFAULT was set to 115200 before? was it redefined elsewhere downstream to 57600?